resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DeletedRuntimeChecker.scala

```diff
@@ -37,28 +37,41 @@ object DeletedRuntimeChecker {
         implicit ev: Ask[F, TraceId]
       ): F[Option[Runtime]] =
         for {
-          clusterOpt <- deps.dataprocService
+          clusterOpt <- deps.runtimeCheckerDeps.dataprocService
             .getCluster(runtime.googleProject, regionName, DataprocClusterName(runtime.runtimeName))
-          _ <- clusterOpt.traverse_ { _ =>
+          isBillingEnabled <- deps.billingService.isBillingEnabled(runtime.googleProject)
```

Comment:
we only need to check `isBillingEnabled ` if cluster still exists in google..can we move this inside `clusterOpt` just that we can save this call in most cases